ragelo/types/types.py

```diff
@@ -222,19 +223,20 @@ class AgentAnswer(Evaluable):
     agent: str
     text: Optional[str] = None
     conversation: Optional[List[ChatMessage]] = None
-    
-    @root_validator
+
+    @validator
```

Comment:
I thought `@validator` is for specific fields while `@root_validator` is for the whole model. Isn't that true?
What's the reason for changing it?

Reply:
The problem is that `@root_validator`would fail if using Pydantic>1. So on lines 9-19 we overload `validator` to be `root_validator` if the user is using pydantic 1 or model_validator in Pydantic 2

Reply:
ah I see, I hadn't noticed the overloading